app/Http/Controllers/ShopController.php

```diff
@@ -4,17 +4,43 @@
 
 use Illuminate\Http\Request;
 use Intervention\Image\Facades\Image;
+use Illuminate\Support\Facades\Storage;
 
 class ShopController extends Controller
 {
     public function store(Request $request)
     {
         $filename = $request->file('photo')->getClientOriginalName();
-        $request->file('photo')->storeAs('shops', $filename);
-
+        $path = $request->file('photo')->storeAs('shops', $filename);
+        
         // TASK: resize the uploaded image from /storage/app/shops/$filename
         //   to size of 500x500 and store it as /storage/app/shops/resized-$filename
         // Use intervention/image package, it's already pre-installed for you
+        try {
+            //code...
+            // dd($request->file('photo')->path().'/'.$filename);
+            // dd($filename);
+            /**** I don't understand why i always get error "Image source not readable" */
+            // $img = Image::make($request->file('photo')) //its not working
+            // $img = Image::make($request->file('photo')->path()) //its also not working
+            // $img = Image::make($request->file('photo')->getRealPath()) //its also not working
+            // $img = Image::make($request->photo) //its also not working
+            // $img = Image::make($path) //its also not working
+            $img = Image::make(Storage::url($path)) //its also not working
+                ->resize(500, 500, function ($constraint) {
+                    $constraint->aspectRatio();
+                })
+                // ->fit(500)
+                ->save(storage_path('shops/'.$filename));
```

Comment:
`Storage::url` will not work since it will not return the full file path, only the relative path e.g **/storage/shops/filename.jpg**. Using `Storage::path` will return the full file path. `storage_path` doesn't append the `app` directory to the path so you either have to add that yourself i.e `storage_path('app/shops/resized-' . $filename)` or use the `Storage` facade that does it for you.

```suggestion
            Image::make(Storage::path($path)) //its also not working
                ->resize(500, 500, function ($constraint) {
                    $constraint->aspectRatio();
                })
                // ->fit(500)
                ->save(Storage::path('shops/resized-' . $filename));
```

Reply:
Ah.. I see 
thanks for the explanation

Reply:
by the way, should we use saved file on storage ?.
I have other project that can call from ```$request->file('photo')->path()``` and that is work (tested on form UI not feature test)
but in this case doesn't work. 🤔

Reply:
> I have other project that can call from `$request->file('photo')->path()`

That will return the full path to the **temporary** file storage, used internally by PHP to store the files before you move them to their final path.

You can test that by running `dd($request->file('photo')->path())`, that should return something along the lines of: */private/var/folders/* and a bunch of random letters. Or whatever the temp path is on your particular system.